### PR TITLE
add configuration to use react-java-mysql sample with Docker Dev Environments feature

### PR DESCRIPTION
**Pull request by glours**
- Add a dedicated Compose file in the `.docker` directory for Dev Env configuration
- Add a new `dev-envs` stage in both `frontend` and `backend` Dockerfiles
- Update documentation
- small changes to `pom.xml` and `compose.yaml`

### CODE REVIEW

File: react-java-mysql/backend/Dockerfile [hunk 1]
```diff
@@ -1,10 +1,30 @@
...
 RUN mvn install
+
+FROM builder as dev-envs
+
+RUN <<EOF
+apt-get update
+apt-get install -y git
+EOF
+
+RUN <<EOF  \
```
Comment (StefanScherer) [inline react-java-mysql/backend/Dockerfile:18]: nit: we probably don‘t need the backslash here

File: react-java-mysql/frontend/Dockerfile [hunk 2]
```diff
@@ -10,12 +12,23 @@ COPY . /code
...
 
+CMD [ "npm", "run", "build" ]
+
+FROM builder AS dev-envs
+RUN <<EOF
+apt-get update
+apt-get install -y git
+EOF
+
+RUN <<EOF  \
```
Comment (StefanScherer) [inline react-java-mysql/frontend/Dockerfile:23]: nit: same here, the backslash

**Review by StefanScherer — COMMENTED** [review]
Only some minor details, otherwise looks good. 😀

File: react-java-mysql/backend/Dockerfile (hunk 1 above)
Comment (glours) [reply react-java-mysql/backend/Dockerfile:18]: My IDE think it smarter than me but it don't know anything about Heredoc notation 🤪

**Comment by glours** [comment]
Arggg I just go a permission errors with the frontend container, need to rework that before merging

**Comment by glours** [comment]
> Arggg I just go a permission errors with the frontend container, need to rework that before merging

Ok that was just me forgetting to run a `npm install` before trying to start the project from the dev container 😅

**Pull request merged**